### PR TITLE
Automatically generate how-tos

### PR DESCRIPTION
**Pull request by fcooper8472**
See #185 

This PR adds a generation python script, and a github actions workflow that will automatically update the website.

### CODE REVIEW

**Review by mirams — DISMISSED** [review]

**Comment by fcooper8472** [comment]
@mirams this is now ready for review, and has generated the following how-tos page:
https://chaste.github.io/docs/how-tos/

#185 has been updated with the details.

File: .github/workflows/update-how-tos.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        run: |
+          python GenerateHowTos.py
+        working-directory: python/infra
+
+      - name: Upload archive files
+        uses: actions/upload-artifact@v3
+        with:
+          name: how-to-markdown
+          path: |
+            chaste_howto_webpage.md
```
Comment (MILeach) [inline .github/workflows/update-how-tos.yml:33]: Just checking this filename/path doesn't want another underscore in `howto`

**Review by MILeach — APPROVED** [review]
Looks good to me!

File: .github/workflows/update-how-tos.yml (hunk 1 above)
Comment (fcooper8472) [reply .github/workflows/update-how-tos.yml:33]: it's just a temp file uploaded as an artifact between jobs - it's actually copied to _index.md when it becomes part of the website.

**Review by mirams — APPROVED** [review]

**Comment by mirams** [comment]
Looks a lot tidier Fergus, thanks. I also very much like the auto-highlighting right hand panel

**Comment by fcooper8472** [comment]
The memory testing failure was only failing to generate the index file due to missing python... this seems like a spurious error, but one that should perhaps be addressed. I'll open a new issue for that.

In the meantime I'll merge this.

**Pull request merged**